src/Microsoft.AspNetCore.Sockets.Common/TransportType.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Sockets
+{
+    [Flags]
+    public enum TransportType
+    {
+        WebSockets = 1,
+        ServerSentEvents = 2,
+        LongPolling = 4,
+        All = WebSockets | ServerSentEvents | LongPolling
```

Comment:
Well, I think it would work fine. The user is choosing which transports to allow the client to use. `All` (or any combination of the flags really) just means "Pick among these transports based on framework-defined logic"